### PR TITLE
[OF-1371] Added ParentId property to SpaceEntity

### PR DESCRIPTION
**Pull request by MAG-mv**
- Added ParentId property to SpaceEntity
- Added serialisation/deserialisation
- Updated serialisation tests to support this
- Correctly constructed EntityLock in all constructors to prevent SerializationTest crashes

### CODE REVIEW

**Review by MAG-IY — DISMISSED** [review]
No comments seems fine. Approved

File: Library/include/CSP/Multiplayer/SpaceEntity.h [hunk 1]
```diff
@@ -341,6 +343,7 @@ class CSP_API SpaceEntity
 	bool IsTransferable;
 	bool IsPersistant;
 	uint64_t OwnerId;
+	uint64_t ParentId;
```
Comment (MAG-SamBirley) [inline Library/include/CSP/Multiplayer/SpaceEntity.h:346]: Observation - we are serialising null for no-parent, but representing in CSP as 0.
Suggestion - Should we align those two things? Perhaps by using csp::Optional<uint64_t> for the type of `ParentId` and using null when there's no parent?

Comment (MAG-mv) [reply Library/include/CSP/Multiplayer/SpaceEntity.h:346]: I'm happy to do that

Comment (MAG-mv) [reply Library/include/CSP/Multiplayer/SpaceEntity.h]: Updated

**Review by MAG-AdamThorn — APPROVED** [review]

**Review by MAG-SamBirley — COMMENTED** [review]
Love the nullable implementation. Reads much nicer. DId we test that it works as expected in C# and web? Not sure we've done an optional member variable before, so might be worth checking before it goes in.

**Comment by MAG-mv** [comment]
> Love the nullable implementation. Reads much nicer. DId we test that it works as expected in C# and web? Not sure we've done an optional member variable before, so might be worth checking before it goes in.

We have now decided to keep the ids private, and no accessors are needed for this ticket

**Review by MAG-SamBirley — APPROVED** [review]

**Pull request merged**